packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -35,8 +35,24 @@ export function useCookie<T = string | null | undefined> (name: string, _opts?:
 
   if (process.client) {
     const callback = () => { writeClientCookie(name, cookie.value, opts as CookieSerializeOptions) }
+
     if (opts.watch) {
-      watch(cookie, callback, { deep: opts.watch !== 'shallow' })
+      const channel = new BroadcastChannel(`nuxt:cookies:${name}`)
+      onUnmounted(() => { channel.close() })
```

Comment:
The test failure is that this needs to run outside of component instance as well...

You could check `getCurrentInstance` to determine if we're in a component.